editor/AnimStatesDialog.cpp

```diff
@@ -591,7 +591,8 @@ void CAnimStatesDialog::UpdateDialog() {
   CheckDlgButton(IDC_GOTO_ROLLING_CHECK, m_anim[mc].elem[AS_GOTO_ALERT_ROLLING].used ? 1 : 0);
   CheckDlgButton(IDC_GOTO_JUMPING_CHECK, m_anim[mc].elem[AS_GOTO_ALERT_JUMPING].used ? 1 : 0);
 
-  for (int i = 0; i < max_states; i++) {
+  int i;
```

Comment:
initialize new `i` in loop every time instead of reusing it, same in other cases. That will reduce coupling between loops, so we can move them around freely when refactoring the code.